python/pyarrow/tests/test_parquet.py

```diff
@@ -1517,14 +1517,9 @@ def test_dataset_read_pandas_common_metadata(tmpdir, preserve_index):
         df.index = pd.Index(np.arange(i * size, (i + 1) * size), name='index')
 
         path = pjoin(dirpath, '{:d}.parquet'.format(i))
-
         table = pa.Table.from_pandas(df, preserve_index=preserve_index)
-
-        # Obliterate metadata
```

Comment:
I'm not sure about deleting this, but this keeps the parts' schema synchronized with `_metadata`

Reply:
I think this should stay here and work. When we have a failure later in `ParquetDatset`, we should add there a flag to the equals comparision to skip the metadata comparision. I would expect that Parquet files have different metadata stored eventhough that they are in the same dataset.

Reply:
Yeah, just thought for backward compatibility.